src/index.ts

```diff
@@ -415,6 +415,36 @@ function shouldIgnore (filename: string, ignore: RegExp[]) {
   return ignore.some(x => x.test(relname))
 }
 
+/**
+ * unsafe re-order and make .ts > .js
+ */
+function registerHandler (
+  opts: Options,
+  extensions: string[],
+  ignore: RegExp[],
+  register: Register,
+  originalJsHandler: (m: NodeModule, filename: string) => any
+) {
+  if (!opts.preferTsExts) {
+    return extensions.forEach(ext => registerExtension(ext, ignore, register, originalJsHandler))
+  }
+
+  // @todo a better way with options
+  ['.ts', '.tsx']
+    .concat(Object.keys(require.extensions), extensions) // tslint:disable-line
+    .filter((element, index, array: string[]) => array.indexOf(element) === index)
```

Comment:
Let's just use `Set`, i.e. `new Set([...extensions, Object.keys(require.extensions)])`.

Reply:
I didn't think Set guaranteed preserving order, which is why I didn't use it.

Reply:
It should based on insertion order. Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set